text/0000-cargo-templates.md

```diff
@@ -0,0 +1,275 @@
+
+#### Yeoman
+
+ [Yeoman](https://yeoman.io/) is a "project template generator that has been in development for years, and has grown to have over 5000 project templates." - @kornel 
+
+We don't need to have all the same features as Yeoman, but there may be lessons to learn form it. 
+
+To scaffold a new project:
+```shell
+yo webapp
+```
+
+To find out the options for the project, pass the `--help` flag:
+```shell
+yo webapp --help
+```
+
+### Python
+
+#### `paster`
+
+It appears there is a `pip` package created by the community called `pastescript` , which, “[creates] file layouts for packages.” You can use it by running:
+
+```
+paster create --template=basic_package MyPackage
+```
+
+### ReasonML
+
+#### `bsb -init my-new-project -theme basic-reason`
+
+Using [BuckleScript](https://bucklescript.github.io/en/), you can [initialize a basic Reason project](https://reasonml.org/docs/manual/latest/installation#new-project). This includes the following: `README.md` `bsconfig.json` `node_modules` `package.json` `src` and is similar to `cargo new` .
+
+### Has the community suggested this before? Are there crates that solve this problem already?
+
+A [similar RFC](https://internals.rust-lang.org/t/pre-rfc-cargo-templates/5056) for cargo templates was written back in April 2017 and shared on the internals forum. It seemed like there were a lot of discussions, but no consensus reached.
+
+The community has also created two crates that solve similar problems:
+
+* [cargo-generate](<https://crates.io/crates/cargo-generate>): a developer tool to help you get up and running quickly with a new Rust project by leveraging a pre-existing git repository as a template
+* [kickstart](https://github.com/Keats/kickstart): created by @Keats described as "A scaffolding tool to get new projects up and running quickly"
+* [cookiecutter](<https://github.com/cookiecutter/cookiecutter>) : not a Rust project, but "A command-line utility that creates projects from cookiecutters (project templates)"
+
+In addition, there was a lot of discussion both from the Cargo team and the community on [this issue](https://github.com/rust-lang/cargo/issues/3506). It has been decided that there [still remains disagreement](https://github.com/rust-lang/cargo/pull/8029#issuecomment-604756599) among both the Cargo team and the community on how to solve this, hence why this RFC seems to be the logical next step.
+
+# What lessons can we learn from what other communities have done here?
+
+There are pros and cons to having templates or some type of template ecosystem. I think the biggest question is **who will maintain them?**It may not be directly related, but it's an important point to consider should templates be added to `cargo new` .
```

Comment:
```suggestion
There are pros and cons to having templates or some type of template ecosystem. I think the biggest question is **who will maintain them?** It may not be directly related, but it's an important point to consider should templates be added to `cargo new` .
```